sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -258,4 +223,47 @@ object SharedState extends Logging {
         throw new IllegalArgumentException(s"Error while instantiating '$className':", e)
     }
   }
+
+  /**
+   * Load hive-site.xml into hadoopConf and determine the warehouse path we want to use, based on
+   * the config from both hive and Spark SQL. Finally set the warehouse config value to sparkConf.
+   */
+  def loadHiveConfFile(
+      sparkConf: SparkConf,
+      hadoopConf: Configuration): Boolean = {
+    val hiveWarehouseKey = "hive.metastore.warehouse.dir"
+    val configFile = Utils.getContextOrSparkClassLoader.getResource("hive-site.xml")
+    if (configFile != null) {
+      logInfo(s"loading hive config file: $configFile")
+      // this call actually will override existing configurations from hive-site.xml
+      hadoopConf.addResource(configFile)
```

Comment:
```java
 /**
   * Add a configuration resource. 
   * 
   * The properties of this resource will override properties of previously 
   * added resources, unless they were marked <a href="#Final">final</a>. 
   * 
   * @param url url of the resource to be added, the local filesystem is 
   *            examined directly to find the resource, without referring to 
   *            the classpath.
   */
```

Reply:
SO we need to restore spark.hadoop(hive).* to avoid being overrided

Reply:
how about we create a fresh hadoop conf first, then load hive-site.xml to it, and finally merge the existing `hadoopConf` into it

Reply:
we need hive-site.xml in sparkContext.hadoopConfiguration not just the new fresh hadoopConf